### PR TITLE
[VF-10] Added groupId security check and unit tests for PostNewslettersRecipients

### PR DESCRIPTION
**Pull request by infantesimone**
#### Description

This PR introduces a new security check on newsletter id (groupId) and some unit tests.

#### List of Changes

- unit tests for `PostNewslettersRecipients`;
- `groupId` security check.
- new env parameter `MAILUP_ALLOWED_GROUPS`.

#### Motivation and Context

The goal of security check on newsletter id  is to allow new recipients only on the newsletters defined in the list `MAILUP_ALLOWED_GROUPS`.

#### How Has This Been Tested?

1. yarn build
2. yarn start
3. call api define in `PostNewslettersRecipients` with groupId not define in `MAILUP_ALLOWED_GROUPS`

Alternatively: 

- `yarn test PostNewslettersRecipients`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      {
+        email,
+        recaptchaToken
+      } as RecipientRequest
+    );
+
+    expect(response.kind).toBe("IResponseErrorInternal");
+  });
+
+  it("should return IResponseErrorInternal if adding receiver to mailup fails fails", async () => {
```
Comment (pasqualespica) [inline io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts:143]: You could remove one `fails` in to desc

File: io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      } as RecipientRequest
+    );
+
+    expect(response.kind).toBe("IResponseErrorInternal");
+  });
+
+  it("should return IResponseErrorForbiddenNotAuthorized if groupId is unauthorized", async () => {
+    jest
+      .spyOn(handlers, "addRecipientToMailupListTask")
+      .mockReturnValueOnce(fromLeft(Error()));
```
Comment (pasqualespica) [inline io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts:181]: I think this mock value is useless , `fromPredicate` test on `groupId` is the firs step in to handler

**Review by pasqualespica — APPROVED** [review]
Great work 👍 
But i think that second step in to PR description should be replaced with something like this `yarn test PostNewslettersRecipients`

File: io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts (hunk 2 above)
Comment (infantesimone) [reply io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts:181]: right 👍

**Pull request merged**